Sources/Core/AWSClientRuntime/AWSServiceSpecificConfiguration.swift

```diff
@@ -31,5 +34,19 @@ public protocol AWSServiceSpecificConfiguration {
     /// Creates a service-specific configuration for this service
     /// - Parameter endpointResolver: An endpoint resolver for the service, or `nil` to let
     /// the service resolve its own endpoint.
-    init(endpointResolver: AWSServiceEndpointResolver?) throws
+    init(
+        endpointResolver: AWSServiceEndpointResolver?,
+        authSchemeResolver: ClientRuntime.AuthSchemeResolver?,
+        authSchemes: [ClientRuntime.AuthScheme]?
+    ) throws
+
+    /// List of configured auth schemes for the service.
+    ///
+    /// If none is provided at compile time, value will default to auth schemes supported by the service as defined in its model.
+    var authSchemes: [ClientRuntime.AuthScheme] { get }
```

Comment:
Why does this property need to be in `ServiceSpecificConfiguration` and not in `AWSClientConfiguration`?

`endpointResolver` is here because every service has its own type of endpoint resolver.  You can see it is of an `associatedtype` of this protocol, `AWSServiceEndpointResolver`, which is determined by the specific AWS service being configured.  That's what makes `endpointResolver` service-specific.

Since `authSchemes` is always of a single type that is defined in Smithy, it would be better placed in `AWSClientConfiguration`.

Same comment goes for `authSchemeResolver` as well.

Reply:
I see, that's a good point. However every service has its own type of auth scheme resolver as well, in the format of `ServiceNameAuthSchemeResolver`. I'll update this so that `authSchemes` is moved to non-service specific config, and have `authSchemeResolver` be an associated type as well.

Reply:
Actually, on second thought, I think both naturally belong in `ServiceSpecificConfiguration` because every service has its own type of auth scheme resolver, and every service has its own unique set of auth schemes configured for it based on the smithy model. And in both cases they need to be code-generated based on the model & the service name.

Reply:
It's not the value that goes in the property that determines where it should go, it's the property itself.  If the property & its type are not specific to the service, it should go on the main config and not on `ServiceSpecificConfiguration`.

Reason is that `ServiceSpecificConfiguration` is code-generated.  So, adding two properties to it adds bloat to all ~350 service-specific configurations when we can instead add it to the one shared config.

Reply:
Each service has its own unique auth scheme resolver type. And each service has a unique set of auth schemes it supports, which has to be code generated from its model. I think given these two conditions they both belong in `ServiceSpecificConfiguration`. - Will discuss further offline

Reply:
After further offline discussion, concluded to move `authSchemes` property to general AWS config, and `authSchemeResolver` to service specific config. Reasoning being there exists service specific resolver _type_ for `authSchemeResolver`, whereas there does not exist service specific type for `authSchemes`.